merlin/models/tf/inputs/embedding.py

```diff
@@ -85,6 +87,32 @@ def from_pretrained(
     def input_dim(self):
         return self.col_schema.int_domain.max + 1
 
+    @property
+    def table_name(self):
+        return self.col_schema.int_domain.name or self.col_schema.name
+
+    def add_feature(self, col_schema: ColumnSchema) -> None:
+        if not col_schema.int_domain:
+            raise ValueError("`col_schema` needs to have an int-domain")
+
+        if (
+            col_schema.int_domain.name
+            and self.col_schema.int_domain.name
+            and col_schema.int_domain.name != self.col_schema.int_domain.name
+        ):
+            raise ValueError(
+                "`col_schema` int-domain name does not match table domain name. "
+                f"{col_schema.int_domain.name} != {self.col_schema.int_domain.name} "
+            )
+
+        if col_schema.int_domain.max != self.col_schema.int_domain.max:
```

Comment:
Shouldn't we be using `cardinality` instead of `int_domain.max`?

Reply:
the implementation of the embedding table uses int_domain.max so I think changing that should be a separate PR if we wanted to do that.

I've been wondering what benefit we'd get from changing to use the `embedding_sizes.cardinality` property. The cardinality makes sense as a property of the data, but doesn't feel like `embedding_sizes` is an attribute of the data depending on what we intend the schema to describe. If the main reason is to improve the code clarity we could perhaps implement a helper property on the columnschema. 

The support for shared embedding tables is based on the int domain name. Which suggests that the int domain is a closer match for computing the cardinality too.